modules/mod-deep-learning/ModelCardPanel.cpp

```diff
@@ -0,0 +1,518 @@
+   {
+      TranslatableString name = Verbatim("%s").Format(tag);
+      SetLabel(name);
+
+      // SetWindowStyle(wxBORDER_SIMPLE);
+      SetBackgroundColour(color);
+
+      ShuttleGui S(this, eIsCreating);
+      wxStaticText *txt = S.AddVariableText(name, true);
+      SetMaxSize(wxSize(90, static_cast<int>(txt->GetSize().GetHeight() * 2)));
+      SetVirtualSize(txt->GetSize());
+
+      txt->SetBackgroundColour(color);
+      wxFont font = txt->GetFont();
+      font.SetPointSize(11);
+      txt->SetFont(font);
+
+      Refresh();
+      
+      Fit();
+      Layout();
+   }
+};
+
+// ModelCardPanel
+
+ModelCardPanel::ModelCardPanel(wxWindow *parent, wxWindowID winid, ModelCardHolder card, 
+                              DeepLearningEffectBase *effect, ModelManagerPanel *managerPanel, const wxSize& size)
+    : wxPanelWrapper(parent, winid, wxDefaultPosition, size, wxBORDER_SIMPLE | wxEXPAND), 
+      mModelName(nullptr),
+      mModelSize(nullptr),
+      mModelAuthor(nullptr), 
+      mShortDescription(nullptr),  
+      mLongDescription(nullptr),  
+      mInstallButton(nullptr), 
+      mInstallStatusText(nullptr),
+      mInstallProgressGauge(nullptr),
+      mSelectButton(nullptr), 
+      mMoreInfoButton(nullptr)
+{
+   SetLabel(XO("Model Card"));
+   SetName(XO("Model Card"));
+
+   mParent = parent;
+   mCard = card;
+   mEffect = effect;
+   mManagerPanel = managerPanel;
+   // TransferDataToWindow();
+}
+
+void ModelCardPanel::Populate()
+{
+   SetAutoLayout(true);
+   ShuttleGui S(this, eIsCreating);
+   PopulateOrExchange(S);
+   Fit();
+   Center();
+   Layout();
+}
+
+void ModelCardPanel::PopulateWithNewCard(ModelCardHolder card)
+{
+   DestroyChildren();
+   SetSizer(nullptr);
+
+   mCard = card;
+   Populate();
+   Refresh();
+   mParent->Fit();
+   mParent->Refresh();
+   mParent->Layout();
+   mParent->GetParent()->Fit();
+   mParent->GetParent()->Refresh();
+   mParent->GetParent()->Layout();
+
+}
+
+bool ModelCardPanel::TransferDataToWindow()
+{
+   return true;
+}
+
+bool ModelCardPanel::TransferDataFromWindow()
+{
+   ShuttleGui S(this, eIsSavingToPrefs);
+   PopulateOrExchange(S);
+
+   gPrefs->Flush();
+
+   return true;
+}
+
+void ModelCardPanel::PopulateNameAndAuthor(ShuttleGui &S)
+{
+   // {repo-name} [model size]
+   // S.StartHorizontalLay(wxALIGN_LEFT, true);
+   S.StartMultiColumn(2, wxALIGN_LEFT);
+   {
+      mModelName = S.AddVariableText(Verbatim("%s")
+                                          .Format(mCard->name()),
+                                       false, wxLEFT);
+      mModelName->SetFont(wxFont(wxFontInfo().Bold()));
+
+      // model size
+      mModelSize = S.AddVariableText(Verbatim("[- MB]"));
+      FetchModelSize();
+
+   }
+   S.EndMultiColumn();
+   // S.EndHorizontalLay();
+
+   // by {author}
+   S.StartHorizontalLay(wxALIGN_LEFT, true);
+   {
+      S.AddVariableText(XC("by", "author of the model"));
```

Comment:
Such a small ambiguous word without context needs its i18n-hint comment that it means "programmed by" someone